File: lib/src/main/java/DTO/helpers/TaskSummaryDTO.java
package DTO.helpers;

/**
 * DTO for the "Task Hours Table".
 * Represents detailed information for a single task relevant to effort tracking.
 */

public record TaskSummaryDTO(
        Long id,
        String taskName,
        String developerName,
        int estimatedHours, // story points for a task
        int actualHours // Logged hours for a task
) {}

File: frontend/src/lib/types/DTO/helpers/TaskSummaryDTO.ts
import {z} from "zod";

export const TaskSummaryItemSchema = z.object({
    id: z.number().int(),
    taskName: z.string(),
    developerName: z.string(),
    estimatedHours: z.number().int(),
    actualHours: z.number().int(),
});

export const TaskSummarySchema = z.array(TaskSummaryItemSchema);
export type TaskSummaryDTO = z.infer<typeof TaskSummarySchema>;
